tensor/stats/stats/funcs.go

```diff
@@ -4,62 +4,341 @@
+// and returns the Float64 output values for subsequent use.
+func CountFuncOut64(in, out *tensor.Indexed) *tensor.Indexed {
+	return VectorizeOut64(NFunc, func(idx int, tsr ...*tensor.Indexed) {
+		VecFunc(idx, tsr[0], tsr[1], 0, func(val, agg float64) float64 {
+			return agg + 1
+		})
+	}, in, out)
 }
 
-// SumFunc is an StatFunc that computes a sum aggregate.
-// use 0 as initial value.
-func SumFunc(idx int, val float64, agg float64) float64 {
-	return agg + val
+// CountFunc computes the count of non-NaN tensor values.
+// See [StatsFunc] for general information.
+func CountFunc(in, out *tensor.Indexed) {
+	CountFuncOut64(in, out)
 }
 
-// Prodfunc is an StatFunc that computes a product aggregate.
-// use 1 as initial value.
-func ProdFunc(idx int, val float64, agg float64) float64 {
-	return agg * val
+// SumFuncOut64 computes the sum of tensor values,
+// and returns the Float64 output values for subsequent use.
+func SumFuncOut64(in, out *tensor.Indexed) *tensor.Indexed {
+	return VectorizeOut64(NFunc, func(idx int, tsr ...*tensor.Indexed) {
+		VecFunc(idx, tsr[0], tsr[1], 0, func(val, agg float64) float64 {
+			return agg + val
+		})
+	}, in, out)
 }
 
-// MinFunc is an StatFunc that computes a min aggregate.
-// use math.MaxFloat64 for initial agg value.
-func MinFunc(idx int, val float64, agg float64) float64 {
-	return math.Min(agg, val)
+// SumFunc computes the sum of tensor values.
+// See [StatsFunc] for general information.
+func SumFunc(in, out *tensor.Indexed) {
+	SumFuncOut64(in, out)
 }
 
-// MaxFunc is an StatFunc that computes a max aggregate.
-// use -math.MaxFloat64 for initial agg value.
-func MaxFunc(idx int, val float64, agg float64) float64 {
-	return math.Max(agg, val)
+// SumAbsFunc computes the sum of absolute-value-of tensor values.
+// This is also known as the L1 norm.
+// See [StatsFunc] for general information.
+func SumAbsFunc(in, out *tensor.Indexed) {
+	VectorizeOut64(NFunc, func(idx int, tsr ...*tensor.Indexed) {
+		VecFunc(idx, tsr[0], tsr[1], 0, func(val, agg float64) float64 {
+			return agg + math.Abs(val)
+		})
+	}, in, out)
 }
 
-// MinAbsFunc is an StatFunc that computes a min aggregate.
-// use math.MaxFloat64 for initial agg value.
-func MinAbsFunc(idx int, val float64, agg float64) float64 {
-	return math.Min(agg, math.Abs(val))
+// ProdFunc computes the product of tensor values.
+// See [StatsFunc] for general information.
+func ProdFunc(in, out *tensor.Indexed) {
+	VectorizeOut64(NFunc, func(idx int, tsr ...*tensor.Indexed) {
+		VecFunc(idx, tsr[0], tsr[1], 1, func(val, agg float64) float64 {
+			return agg * val
+		})
+	}, in, out)
 }
 
-// MaxAbsFunc is an StatFunc that computes a max aggregate.
-// use -math.MaxFloat64 for initial agg value.
-func MaxAbsFunc(idx int, val float64, agg float64) float64 {
-	return math.Max(agg, math.Abs(val))
+// MinFunc computes the min of tensor values.
+// See [StatsFunc] for general information.
+func MinFunc(in, out *tensor.Indexed) {
```

Comment:
Still not a fan of all of the `Func` suffixes.

Reply:
better idea?  enum takes precidence.